### ISSUE TITLE
APIGateway CR deletion support

### ISSUE THREAD

**Issue by strekm** [kyma-project/api-gateway#559]
**Description**

API Gateway controller should react on APIGateway CR deletion event. If there are no VS/APIRule referencing default gateway, controller should remove default gateway. In case there are any Istio VS/APIRule referencing default gateway these resources should be listed in logs and APIGateway CR should be in warning status indicating user action. Once user will delete or modify logged resources default gateway can be removed safely and deletion of APIGateway CR can continue.

ACs:
- [x] user can delete APIGateway CR
- [x] APIGateway CR is in warning state if there are any resources using default gateway
- [x] resources blocking deletion listed in logs
- [x] default gateway removed
- [x] adapted deletion conditions:
* Kyma Gateway is blocked by referring APIRules and VirtualServices
* API-Gateway CR is blocked by any APIRules on the cluster
- [x] [godog integration test/scenario/feature](https://github.com/kyma-project/api-gateway/pull/567/commits/9a7d8e3de109f87f7dac1f5e9f1d8f80121c8ce9) implemented

**Reasons**

api-gateway modularisation

**DoD:**
- [x] provide documentation
~~- [ ] release notes and What's New updates for Kyma customers~~
- [x] provide unit tests
- [x] provide integration tests
- [x] test on production-like environment
~~- [ ] verify resource limits~~
~~- [ ] followup issue~~
~~- [ ] create release and bump in kyma~~
- [x] PR reviewer will verify code coverage and evaluate if it is acceptable

**Attachments**
part of: https://github.com/kyma-project/api-gateway/issues/130

**PRs**
- [x] https://github.com/kyma-project/api-gateway/pull/567
